internal/cloudapi/server.go

```diff
@@ -225,3 +232,23 @@ func (server *Server) ComposeStatus(w http.ResponseWriter, r *http.Request, id s
 		panic("Failed to write response")
 	}
 }
+
+func composeStatusFromJobStatus(js *worker.JobStatus) string {
+	if js.Canceled {
+		return StatusFailure
```

Comment:
Should we introduce a `cancelled` state? Seems useful. Then again, I guess we do not yet support cancelling composes from this API.

Reply:
This would be a future improvement.

Reply:
Agreed, let's not muddy this PR even more. I very much like the idea of such a state, though.